File: ui/src/lib/api/agent-memory.ts
import { del, get } from "./fetch";

// ==================== Agent 长期记忆类型（ADR 004 跨会话记忆） ====================

/** 单条 Agent 长期记忆 */
export type AgentMemoryItem = {
  id: string;
  content: string;
  source_session_id: string | null;
  created_at: string;
};

/** Agent 长期记忆列表 */
export type AgentMemoryList = {
  items: AgentMemoryItem[];
  total: number;
};

// ==================== Agent 长期记忆 API ====================

export const agentMemoryApi = {
  /** 列出当前组织的 Agent 长期记忆（按时间倒序） */
  list: (): Promise<AgentMemoryList> => {
    return get<AgentMemoryList>("/agent-memories");
  },

  /** 删除一条 Agent 长期记忆 */
  remove: (memoryId: string): Promise<unknown> => {
    return del(`/agent-memories/${memoryId}`);
  },
};

File: api/app/interfaces/schemas/agent_memory.py
"""Agent 长期记忆接口 schema（ADR 004）。"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel

from app.domain.models.agent_memory import AgentMemory


class AgentMemoryItem(BaseModel):
    """单条长期记忆的对外表示。"""
    id: str
    content: str
    source_session_id: Optional[str] = None
    created_at: datetime

    @classmethod
    def from_domain(cls, memory: AgentMemory) -> "AgentMemoryItem":
        return cls(
            id=memory.id,
            content=memory.content,
            source_session_id=memory.source_session_id,
            created_at=memory.created_at,
        )


class AgentMemoryListResponse(BaseModel):
    """长期记忆列表响应。"""
    items: List[AgentMemoryItem]
    total: int

    @classmethod
    def from_domain(cls, memories: List[AgentMemory]) -> "AgentMemoryListResponse":
        items = [AgentMemoryItem.from_domain(m) for m in memories]
        return cls(items=items, total=len(items))
